src/shell/shell_misc.cpp

```diff
@@ -506,8 +513,13 @@ bool DOS_Shell::Execute(char * name,char * args) {
 				while ( *p == 0 && (p-parseline) < 250) p++; //Skip empty fields
 				if ((p-parseline) < 250) { //Found something. Lets get the first letter and break it up
 					p++;
-					memmove(static_cast<void*>(p + 1),static_cast<void*>(p),(250-(p-parseline)));
-					if ((p-parseline) < 250) *p = 0;
+					const auto tail_length = 250 - (p - parseline);
+					assert(tail_length >= 0); // ensure this won't wrap-around when cast to unsigned
+					memmove(static_cast<void *>(p + 1),
+					        static_cast<void *>(p),
```

Comment:
You never need to cast pointers to `void*`. Don't do it, please.

Original code was bad by using casts here, it should have been written simply: `memmove(p + 1, p, tail_length);` - more readable and more correct.